### ISSUE TITLE
6.15 WithdrawStakeRewards Growing Gas Cost

### ISSUE THREAD

**Issue by renlulu** [Zilliqa/staking-contract#140]
Transition `WithdrawStakeRewards` calls procedure `WithdrawalStakeRewards`.

```
procedure WithdrawalStakeRewards(deleg: ByStr20, ssn_operator: ByStr20)
last_withdraw_cycle_deleg_m <- last_withdraw_cycle_deleg[deleg][ssn_operator];
last_withdraw_cycle = option_uint32_value last_withdraw_cycle_deleg_m;
lrc <- lastrewardcycle;
m = builtin add last_withdraw_cycle uint32_one;
n = builtin add lrc uint32_one;
list_need_compute_rewards = iota m n;
...
```

Then the procedure `CalcStakeDelegPerCycle` is applied to all entries of `list_need_compute_rewards`. If a delegator doesn't withdraw his rewards, each reward cycle will increase the number of entries in this list. Hence the gas cost of this computation will increase with time. In the worst case this may eventually block the delegator from successfully withdrawing his reward due to
high gas costs.

**Comment by renlulu** [comment]
I think we roughly have some idea about gas fee cost, we have done several around of gas simulation. We need do a final gas simulation once everything is ready including smart contracts and testnet (with gas fee reduction).

**Comment by renlulu** [comment]
Won't fix.
